model-api/src/commonMain/kotlin/org/modelix/model/api/INode.kt

```diff
@@ -222,6 +222,7 @@ interface INode {
     fun moveChild(role: IChildLink, index: Int, child: INode) = moveChild(role.key(this), index, child)
     fun addNewChild(role: IChildLink, index: Int, concept: IConcept?): INode = addNewChild(role.key(this), index, concept)
     fun addNewChild(role: IChildLink, index: Int, concept: IConceptReference?): INode = addNewChild(role.key(this), index, concept)
+    fun addNewChild(role: IChildLink, index: Int, templateNode: INode): INode = addNewChild(role, index, templateNode.concept)
```

Comment:
Can you add some documentation on what this method does? This is not straightforward to understand.

Reply:
Added documentation.